metrics/collectors/images/images_collector.py

```diff
@@ -0,0 +1,111 @@
+        super().__init__(dry_run, verbose)
+
+        self.lp = Launchpad.login_anonymously(
+            "metrics",
+            "production",
+            launchpadlib_dir=tempfile.mkdtemp(),
+            version="devel",
+        )
+        self.ubuntu = self.lp.distributions["ubuntu"]
+        self.active_series = {s.name: s for s in self.ubuntu.series if s.active}
+        self.date_now = datetime.datetime.now()
+
+    def rsync_list_images(self):
+        rsync = ""
+        for url in RSYNC_SERVER_REQUESTS:
+            for img in IMAGE_FORMATS:
+                try:
+                    self.log.debug("Rsync listing %s" % url + img)
+                    rsync += subprocess.check_output(
+                        [
+                            "rsync",
+                            "-4",
+                            "--dry-run",
+                            "-RL",
+                            "--out-format='%l %M %f'",
+                            "--archive",
+                            "%s" % url + img,
+                            "/tmp",
+                        ],
+                        text=True,
+                    )
+                except subprocess.CalledProcessError as e:
+                    self.log.error("rsync call failed: %s", e.output)
+        return rsync
+
+    def collect(self):
+        """ Collect the daily images details"""
+        data = []
+
+        rsync_cmd_output = self.rsync_list_images()
+
+        for rsyncline in rsync_cmd_output.splitlines():
+            if not any(ext in rsyncline for ext in IMAGE_FORMATS):
+                continue
+            if "current" in rsyncline or "pending" in rsyncline:
+                # the format is specific in the rsync call '%l %M %f'
+                size, mtime, path = rsyncline.strip("'").split()
+                date_current_image = datetime.datetime.strptime(
+                    mtime, "%Y/%m/%d-%H:%M:%S"
+                )
+                image_age = (self.date_now - date_current_image).days
+
+                # there are variations of the naming scheme so guess a bit
+                path_table = path.split("/")
+                # the path starts with a serie name then it's a desktop iso
+                if path_table[0] in self.active_series:
+                    flavor = "ubuntu"
+                # or with 'daily-live'
+                elif path_table[0] == "daily-live":
+                    flavor = "ubuntu"
+                # otherwise it starts with the flavor
+                else:
+                    flavor = path_table[0]
+
+                # the path ends with the image filename
+                image_name = path_table[-1]
+
+                # let's filter out old ubuntu-core-16 images
+                if image_name.startswith("ubuntu-core-16"):
+                    continue
+
+                if "current" in rsyncline:
+                    current_or_pending = "current"
+                else:
+                    current_or_pending = "daily"
```

Comment:
I think I want this string to say `pending` instead of `daily` as well as the variable :upside_down_face:.

Reply:
Indeed, it's another error from my part, I renamed the variable and didn't pay attention, thanks for catching it and fixing!